Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,64 @@
-Add your answers to the Algorithms exercises here.
\ No newline at end of file
+Add your answers to the Algorithms exercises here.
+
+
+A ) the code would never end. the value of n would accelerate away from
+"a" linearly i believe. the while loop could be considered n^3 and the inner loop 
+its n^2+a which if subtracted reaches O(n) now I also have to consider "a" so maybe the time complexity is O(n + a)?  
+
+
+B) I think the time complexity of this function is O(n^4) 
+a normal double for loop is o(n^2) using this I cam to my conclusion 
+
+for i in range n,   length of n  
+        increase i 
+        for j in range i + 1, to n    
+        <!-- range is still determined by n but could be getting smaller because they increase the value of i by one -->
+            add one to j
+            for k in range k+ 1, to n      
+            <!-- range n is still being approached but this time k times k in being incrase by one for some reason -->
+                for l in range k + 1 to k + 10     
+                <!-- l is limited by K instead of n, k has a hard limit of k+10 instead of n times              -->
+                    increase l by one             
+                    <!-- loop runs until until k runs out then incrases sum by one. -->
+                    increase sum by 1
+
+ 
+
+
+C) bunnies looks to be a number of bunnies. if no bunnies exist return 0
+if not then return double the amount of bunnes inputed 
+2 + O(n-1) time complexity?
+
+example 5 bunnies
+2 + (input 4) - returns 2 + (input 3) - returns 2 + (input 2) - returns 2 + (input 1) - returns 2 + (input 0 ) - returns 0
+
+2 + 2 + 2 + 2 + 2 = 10 on an input of 5 
```

Comment:
Even based on this, if you were to plug in 6,7,8 you would see that the answers would be growing by two each time, linearly.